gax/src/main/java/com/google/api/gax/core/Distribution.java

```diff
@@ -31,128 +31,75 @@
 
 import com.google.api.core.BetaApi;
 import com.google.common.base.MoreObjects;
-import com.google.common.base.MoreObjects.ToStringHelper;
 import com.google.common.base.Preconditions;
-import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLongArray;
 
 /**
- * Takes measurements and stores them in linear buckets from 0 to totalBuckets - 1, along with
- * utilities to calculate percentiles for analysis of results.
+ * Distribution records values from {@code 0} (inclusive) to {@code maxValue} (exclusive) and
+ * computes their percentiles.
+ *
+ * <p>Methods may be called concurrently.
  */
 @BetaApi
 public class Distribution {
 
-  private final AtomicLong[] bucketCounts;
-  private long count;
-  private double mean;
-  private double sumOfSquaredDeviation;
+  private final AtomicLongArray buckets;
+  private final AtomicInteger count = new AtomicInteger(0);
 
-  public Distribution(int totalBuckets) {
-    Preconditions.checkArgument(totalBuckets > 0);
-    bucketCounts = new AtomicLong[totalBuckets];
-    for (int i = 0; i < totalBuckets; ++i) {
-      bucketCounts[i] = new AtomicLong();
-    }
+  public Distribution(int maxValue) {
+    Preconditions.checkArgument(maxValue > 0);
+    buckets = new AtomicLongArray(maxValue);
   }
 
-  /** Get the bucket that records values up to the given percentile. */
-  public long getNthPercentile(double percentile) {
+  /**
+   * Get the percentile of recorded values. If called concurrently with {@link record(int)}, the
+   * result is an approximate.
+   */
+  public int getNthPercentile(double percentile) {
```

Comment:
Changed to `int`. It's useless to return `long` since we don't allow users to record `long` values.